### ISSUE TITLE
Fix false positives for escaped slashes in string-no-newline

### ISSUE THREAD

**Issue by Vinnl** [stylelint/stylelint#2796]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug. Foundation 6.4 added the new setting `$breadcrumbs-item-separator-item-rtl: '\\';`. This appears to trip up stylelint, which emits a warning for that line and what appears to be every line after it saying:

>  275:49  ✖  Unexpected newline in string                                string-no-newline

Ignoring the line as follows didn't help:

    $breadcrumbs-item-separator-item-rtl: '\\'; /* stylelint-disable-line string-no-newline */

> Which rule, if any, is this issue related to?

I guess `string-no-newline`?

> What CSS is needed to reproduce this issue?

```scss
  $breadcrumbs-item-separator-item-rtl: '\\'; /* stylelint-disable-line string-no-newline */
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "extends": "stylelint-config-standard"
}
```

> Which version of stylelint are you using?

8.0.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS variables.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings was flagged:

```shell
>  275:49  ✖  Unexpected newline in string                                string-no-newline
```

**Comment by Vinnl** [comment]
I tried to my [a fork](https://github.com/Vinnl/stylelint/tree/2796-testcase) with a failing test case (see https://github.com/Vinnl/stylelint/commit/b045bf9f5878d80c0785ed4e0f244d7cf857974f), and it did fail, but unfortunately not on the error I'm trying to show:

```
    - Expected
    + Received

    -Array []
    +Array [
    +  Object {
    +    "column": 7,
    +    "line": 1,
    +    "rule": "CssSyntaxError",
    +    "severity": "error",
    +    "text": "Unclosed string (CssSyntaxError)",
    +  },
    +]
```

**Comment by evilebottnawi** [comment]
@Vinnl very instresting, thanks for issue, in near future i investigate this

**Comment by mishelen** [comment]
Confirm that I have same issue (`$breadcrumbs-item-separator-item-rtl: '\\';`) with foundation:

```
"stylelint": "^8.1.1",
"stylelint-bare-webpack-plugin": "^0.1.3",
"stylelint-config-sass-guidelines": "^3.1.0",
"stylelint-config-standard": "^17.0.0",
"stylelint-scss": "^2.1.0"
```

**Comment by jeddy3** [comment]
@Vinnl and @mishelen Thanks for the reports.

I can reproduce this with standard CSS too. The following CSS:

```css
/* stylelint-disable string-no-newline */

a {
  content: "\\";
}

/* stylelint-enable string-no-newline */
```

Producing the following violation:

```
8:1 error Unexpected newline in string (string-no-newline)
```

(Note the error is on line _eight_).

I'm labeling as a bug and help wanted. Please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

**Comment by johnnytomcat** [comment]
In your foundation _settings change 
```scss
$breadcrumbs-item-separator-item: '\002F';
$breadcrumbs-item-separator-item-rtl: '\005C';
```

**Comment by evilebottnawi** [comment]
/cc @Vinnl @mishelen @jeddy3 @johnnytomcat  

Can't reproduced problem :confused: Somebody can check again?

**Comment by jeddy3** [comment]
I also can no longer reproduce the error :)

It was likely fixed when the rule [was refactored to use the selector and value parsers](https://github.com/stylelint/stylelint/commit/68e3b7f3180bef35532712721aca5edd3e4de63b).
